distribution/lib/Standard/Database/0.0.0-dev/src/DB_Table.enso

```diff
@@ -137,9 +138,11 @@ type DB_Table
        - selector: The name or index of the column to get.
     @selector Widget_Helpers.make_column_name_selector
     at : Integer | Text -> DB_Column ! No_Such_Column | Index_Out_Of_Bounds
-    at self (selector:(Integer | Text)=0) = case selector of
-        _ : Integer -> self.make_column (self.internal_columns.at selector)
-        _ -> self.get selector (Error.throw (No_Such_Column.Error selector))
+    at self (selector:(Integer | Text)=0) = 
+        if self.connection.dialect.is_feature_supported Feature.Column_operations . not then Error.throw (Unsupported_Database_Operation.Error "at") else
```

Comment:
Could we add a method to Feature:

```
    if_supported self dialect:Any name:Text=self.to_text ~action = if dialect.is_feature_support self then action else
        Error.throw (Unsupported_Database_Operation.Error name)
```

The methods then become like:

```
    at self (selector:(Integer | Text)=0) = Feature.Column_operations.if_supported self.connection.dialect "at" <|
            case selector of
                _ : Integer -> self.make_column (self.internal_columns.at selector)
                _ -> self.get selector (Error.throw (No_Such_Column.Error selector))
```

Reply:
+1 This would be a useful shorthand

Reply:
Will bring back at in the next MR